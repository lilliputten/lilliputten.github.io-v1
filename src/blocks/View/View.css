// vim: ft=stylus

.View {

  flex: auto;

  display: flex;

  // Top margin for contant page by default
  margin-top: $(NavbarHeight)px;
  transition: opacity .3s ease-out, margin-top .75s ease-out;

  background-color: $(pageBgColor);
  z-index: 5;

  opacity: 0;

  // &_mode_loading,
  &_mode_default,
  &_mode_content,
  &_mode_error {
    opacity: 1;
  }

  &_mode_unloading {
    opacity: .2;
  }

  &_mode_default,
  &_mode_loading {
    margin-top: 0;
  }

  // cubic-bezier(.29, 1.01, 1, -0.68);
}
