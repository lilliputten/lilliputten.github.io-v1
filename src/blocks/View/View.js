/**
 * @module View
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.02.26, 02:42
 * @version 2018.03.19, 02:53
 *
 * TODO:
 *
 *  - 2018.03.11, 04:42 -- Move pages loading & parsing to separate modules.
 *  - 2018.02.24, 03:22 -- Dynamically load file on state update.
 */

// import React, { Fragment } from 'react'
import { decl } from 'bem-react-core'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import 'm:mode=ready|loading|error'

import 'm:hashChange'

import PageLoader from 'libs/PageLoader'

const View_proto = /** @lends View.prototype */{

  block: 'View',

  /** willInit ** {{{ */
  willInit() {

    this.__base.apply(this, arguments);

    // Create page loader
    this.pageLoader = new PageLoader();

    this.state = {
      mode : 'loading',
    };

    // ??? Subscribe to store for page changing... (Correct redux method to update?)
    this.props.store.subscribe(() => {
      const state = this.props.store.getState();
      const pageId = state.page.page;
      this.placePage(pageId);
    });

  },/*}}}*/

  /** componentWillMount ** {{{ */
  componentWillMount() {

    this.__base.apply(this, arguments);

    // Place initial page (default or saved)
    this.placePage();

  },/*}}}*/

  // /** componentDidMount ** {{{ */
  // componentDidMount() {
  //   this.__base.apply(this, arguments);
  //   // DEBUG: Finding App component
  //   // this.App = reactTools.findParentBlock(this, 'App');
  //   console.log('View componentDidMount');
  // },/*}}}*/
  // /** componentDidUpdate ** {{{ */
  // componentDidUpdate() {
  //   this.__base.apply(this, arguments);
  //   console.log('View componentDidUpdate');
  // },/*}}}*/

  /** placePage ** {{{ Update page content on url changing
   * @param {String} [pageId] - Url to show
   * @return {Promise}
   */
  placePage(pageId) {

    if (pageId == null ) {
      pageId = this.props.page; // pageId || config.site.defaultPage;
    }

    // Mode: loading
    this.setState({ mode : 'loading' });

    // DEBUG
    console.log('View:placePage', pageId);

    // Load or resolve page data...
    this.pageLoader.resolve(pageId)

      // Set page data...
      .then(data => {
        this.setState({
          pageId : pageId,
          mode : 'ready',
          attributes: data.attributes,
          html : data.html,
        });
      })

      // Error...
      .catch(err => {
        console.error(err);
        /*DEBUG*/debugger;
        this.setState({
          mode : 'error',
          error : err,
        });
      })

    ;

  },/*}}}*/

  /** mods ** {{{ Modifiers... */
  mods(self) {
    return { ...self.mods,
      DEBUG : this.state.DEBUG,
      mode : this.state.mode,
    };
  },/*}}}*/

}

/** View_static ** {{{ */
const View_static = /* @lends View */{

  propTypes : {
    dispatch : PropTypes.func.isRequired,
    page : PropTypes.string.isRequired,
  },

}/*}}}*/

function mapStateToProps(state) {
  const { page } = state.page || {};
  return { page };
}
export default decl(View_proto, View_static, connect(mapStateToProps));

