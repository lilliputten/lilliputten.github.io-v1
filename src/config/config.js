/**
 * @module config
 * @overview Application config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.10, 02:05
 * @version 2018.03.10, 23:02
 */

import site from './site'

const testMode = (typeof jest === 'object');
const hotMode = !!(typeof module === 'object' && module.hot && module.hot.active);
const DEBUG = testMode ? 'test' : hotMode;

const config = /** @lends config */{

  /** Is test mode? */
  testMode : testMode,
  /** Is hot dev-server mode? */
  hotMode : hotMode,
  /** Is DEBUG mode? (webpack server in hot mode) */
  DEBUG : DEBUG,

  // App start time (DEBUG?)
  startTime : Date.now(), // ( DEBUG === 'test' ) ? 0 : Date.now(),

  /** Site properties */
  site : site,

};

export default config;

