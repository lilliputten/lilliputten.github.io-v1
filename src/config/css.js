/**
 * @module css
 * @overview Export configuration for system styles
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.22, 23:39
 * @version 2018.03.22, 23:41
 */

const primaryColor = '#c10';
const defaultFontSize = 16;

/** cssVariables ** {{{ */
const cssVariables = {

  primaryColor,

  // DEBUG...

  testGlobalColor: 'orange', // test vatiable

  testViewportWidth: 300, // test media size

  // Fonts...

  defaultFontSize,
  fontSize: defaultFontSize,
  fontSizeSm: defaultFontSize - 2,
  fontSizeXs: defaultFontSize - 4,
  fontSizeLg: defaultFontSize + 2,
  fontSizeXl: defaultFontSize + 4,
  fontSizeXxl: defaultFontSize + 6,

  defaultFontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif',
  defaultFontWeight: 400,
  defaultLineHeight: 1.5, // 1.5 ??? too loose? (from mdbootstrap)

  defaultTitleFontSize: 28,
  defaultTitleFontWeight: 300,

  // Page...

  pageBgColor: '#fff',

  defaultTextColor: '#333',
  defaultLinkColor: primaryColor,
  defaultHoveredLinkColor: '#f10',

  // Navbar...

  NavbarHeight: 50,

  NavBarTreshold: 600,

};/*}}}*/

// Using classic export due to requires in `config-override.js`
module.exports = cssVariables;
