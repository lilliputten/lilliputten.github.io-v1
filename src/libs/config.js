/**
 * @module config
 * @overview Application config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.10, 02:05
 * @version 2018.03.10, 23:02
 */

const testMode = (typeof jest === 'object');
const hotMode = !!(typeof module === 'object' && module.hot && module.hot.active);
const DEBUG = testMode ? 'test' : hotMode;

const config = {

  // Test mode?
  testMode : testMode,
  // Hot dev-server mode?
  hotMode : hotMode,
  // Development mode? (webpack server in hot mode)
  DEBUG : DEBUG,

  // Site content root url prefix
  siteRootPrefix : '/site',

  // App start time
  startTime : Date.now(), // ( DEBUG === 'test' ) ? 0 : Date.now(),

};

export default config;

