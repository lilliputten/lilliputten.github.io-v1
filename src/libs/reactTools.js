/**
 * @module reactTools
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.10, 03:56
 * @version 2018.03.10, 04:44
 *
 * TODO:
 *
 * - 2018.03.10, 19:18 -- Find components with modName/modVal & elems.
 * - 2018.03.10, 23:04 -- Move DOM methods to `react-traverse`?
 *
 */

import React from 'react';
import ReactDom from 'react-dom';
import jquery from 'jquery';

const isArray = Array.isArray

const reactTools = {

  // Traverse:

  /** findParentBlock ** {{{ Find closest parent component
   * @param {ReactObject} comp
   * @param {String} compClassName
   * @return {ReactObject}
   * @see [Element.closest() - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)
   * @see [Document.getElementsByClassName() - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByClassName)
   * @see [ReactDOM - React](https://reactjs.org/docs/react-dom.html#finddomnode)
   */
  findParentBlock : function (comp, compClassName) {
    const selector = '.' + compClassName
    const foundDom = this.applyQuerySelector(comp, 'closest', selector)
    const foundComp = foundDom && this.getDomComponent(foundDom)
    return foundComp;
  },/*}}}*/
  /** findChildBlock ** {{{ Find closest parent component
   * @param {ReactObject} comp
   * @param {String} compClassName
   * @return {ReactObject}
   * @see [Element.closest() - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)
   * @see [Document.getElementsByClassName() - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByClassName)
   * @see [ReactDOM - React](https://reactjs.org/docs/react-dom.html#finddomnode)
   */
  findChildBlock : function (comp, compClassName) {
    const selector = '.' + compClassName
    const foundDom = this.applyQuerySelector(comp, 'find', selector)
    const foundComp = foundDom && this.getDomComponent(foundDom)
    return foundComp;
  },/*}}}*/

  // DOM:

  /** applyQuerySelector ** {{{ Apply querySelector lookup method
   * @param {Enzyme:ReactWrapper|ReactComponent|jQueryCollection} target
   * @param {String} op - querySelector operator (eg, `find`, `closest`)
   * @param {String} selector - dom selector (`.className`)
   * @return {jQueryCollection|null)
   */
  applyQuerySelector : function (target, op, selector) {
    let dom = this.getComponentDom(target) // (typeof target.getDOMNode === 'function') ? target.getDOMNode : (target instanceof React.Component) ? ReactDom.findDOMNode(target) : target
    const isJQuery = dom instanceof jquery;
    if (!isJQuery) {
      dom = jquery(dom)
    }
    if (typeof dom[op] === 'function') {
      return dom[op].call(dom, selector)
    }
  },/*}}}*/

  /** getComponentDom ** {{{ Get DOM Node for given component
   * @param {Enzyme:ReactWrapper|ReactComponent|jQueryCollection} target
   * @return {DomObject}
   */
  getComponentDom : function (target) {
    let result = target;
    if (typeof target.getDOMNode === 'function') {
      result = target.getDOMNode();
    }
    else if (target instanceof React.Component) {
      result = ReactDom.findDOMNode(target);
    }
    return result;
  },/*}}}*/
  /** getDomComponent ** {{{ Get react component instance for given DOM node
   * @param {DomObject} dom
   * @return {ReactObject}
   * @see [javascript - React - get React component from a child DOM element? - Stack Overflow](https://stackoverflow.com/questions/24462679/react-get-react-component-from-a-child-dom-element)
   */
  getDomComponent : function (dom) {
    if (dom && dom != null && typeof dom === 'object') {
      // Array|Collection...
      if (isArray(dom) || dom instanceof jquery) {
        dom = dom[0];
      }
      // enzyme ReactWrapper...
      if (typeof dom.instance === 'function') {
        dom = dom.instance();
      }
      if (typeof dom === 'object') {
        for (let key in dom) {
          if (key.startsWith('__reactInternalInstance')) {
            const domItem = dom[key]
            // Old (stackoverflow) code (react15.3.0): `comp = domItem._currentElement._owner._instance`
            // NOTE 2018.03.10, 04:44 -- Works with react16
            const comp = domItem && domItem.return && domItem.return.stateNode
            return comp;
          }
        }
      }
    }
  },/*}}}*/

  // url:

  /** getRelativeUrl ** {{{ Get url relative to current document
   * @param {String} url
   * @return {String}
   */
  getRelativeUrl : function (url) {
    if (typeof window === 'object' && window.location && window.location.origin && url.startsWith(window.location.origin)) {
      const len = window.location.origin.length;
      url = url.substr(len);
    }
    return url;
  },/*}}}*/

  // utils:

  /** delay ** {{{ DEBUG: Timeout
   * TODO: Move to `commonTools`
   * @param {Number} [timeout=1000]
   * @param {*} [data]
   * @return {Promise}
   */
  delay(timeout=1000, data){
    return new Promise(function(resolve, reject){
      setTimeout(function(){
        resolve(data);
      }, timeout)
    });
  },/*}}}*/


};

export default reactTools;

