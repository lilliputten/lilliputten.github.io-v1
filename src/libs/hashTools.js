/**
 * @module hashTools
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.11, 20:32
 * @version 2018.03.11, 20:32
 */

import config from 'config';

const hashTools = {

  /** getFromWindow ** {{{ Get hash from window location */
  getFromWindow : function () {
    return ( typeof window === 'object' && window.location && window.location.hash ) || '';
  },/*}}}*/

  /** setToWindow ** {{{ Set hash to window location
   * @param {String} hash
   */
  setToWindow : function (hash) {
    if ( typeof window === 'object' && window.location ) {
      window.location.hash = hash;
    }
  },/*}}}*/

  /** getPageIdFromWindow ** {{{ Get page id from window location hash
   * @return {String}
   */
  getPageIdFromWindow : function () {
    const hash = this.getFromWindow();
    if ( this.isCorrectHash(hash) ) {
      return this.getPageId(hash);
    }
    return '';
  },/*}}}*/

  /** isCorrecthash ** {{{ Check for correct hash value (empty or in form `#!string/hash`)
   * @param {String} hash
   * @return {Boolean}
   */
  isCorrectHash : function (hash) {

    return !!( hash != null && ( hash === '' || hash.startsWith('#!') || hash.startsWith('#') ) );

  },/*}}}*/

  /** toPageId ** {{{ Convert hash to page id (w/o root & extension)
   * @param {String} hash
   * @return {String} pageId
   */
  toPageId : function (hash) {

    // Strip leading '#!'
    if (hash.startsWith('#')) {
      hash = hash.substr(1);
    }
    // ??
    if (hash.startsWith('!')) {
      hash = hash.substr(1);
    }
    // ???
    if (hash.startsWith('/')) {
      hash = hash.substr(1);
    }

    return hash;

  },/*}}}*/

  /** toUrl ** {{{ Convert hash to base url (w/o extension)
   * @param {String} hash
   * @param {String} [extension]
   * @return {String} url
   */
  toUrl : function (hash, extension) {

    let page = this.toPageId(hash);

    // Adding leading slash
    if (!page.startsWith('/')) {
      page = '/' + page;
    }

    // Adding index if ends with '/'
    if (page.endsWith('/')) {
      page += config.site.defaultIndex;
    }

    // Adding extension
    if (extension) {
      page += extension;
    }

    return config.site.rootPrefix + page;

  },/*}}}*/

  /** fromUrl ** {{{ Convert url to hash
   * @param {String} url
   * @return {String} hash
   */
  fromUrl : function (url) {

    let hash = url;

    // Hash at root url
    if (hash.startsWith('/#!')) {
      return hash.substr(1);
    }

    // Removing default extensions and index
    const stripAtEnd = config.site.defaultExts.concat(config.site.defaultIndex);
    stripAtEnd.map((tail) => {
      if (hash.endsWith(tail)) {
        hash = hash.substr(0, hash.length - tail.length);
      }
      return hash;
    });

    // Removing root prefix
    let found = false;
    if (hash.startsWith(config.site.rootPrefix)) {
      hash = hash.substr(config.site.rootPrefix.length);
      found = true;
    }
    // Removing leading slash
    if (hash.startsWith('/')) {
      hash = hash.substr(1);
      found = true;
    }
    // Invalid url
    if (!found) {
      return null;
    }

    if (this.isCorrectHash(hash)){
      hash = this.toPageId(hash);
    }

    // Return empty value if default page
    if (hash === config.site.defaultPage) {
      hash = '';
    }

    return hash && '#!' + hash;

  },/*}}}*/

  /** getPageId ** {{{ Get page id from hash/url
   * @param {String} urlOrHash - Hash (#!/dir/page) or url (<rootPrefix>/dire/page)
   * NOTE: Highest level method on `toPageId`.
   * @return {null|String) - Return null if can't fing page id.
   */
  getPageId : function (urlOrHash) {

    if (urlOrHash == null) {
      return null;
    }
    else if (urlOrHash === '') {
      return config.site.defaultPage;
    }
    else {
      if (urlOrHash.startsWith(config.site.rootPrefix)) {
        urlOrHash = this.fromUrl(urlOrHash);
      }
      if (this.isCorrectHash(urlOrHash)) {
        const pageId = this.toPageId(urlOrHash);
        if (!pageId || pageId === config.site.defaultPage) {
          return '';
        }
        return pageId;
      }
    }

    return null;

  },/*}}}*/

  /** fromPageId ** {{{ Get hash for pageId
   * @param {String} pageId
   * @return {String} hash
   */
  fromPageId : function (pageId) {
    return '#!' + pageId;
  },/*}}}*/

  /** isDefaultHash ** {{{ Is hash for default page?
   * @param {String} hash - Hash (#!/dir/page)
   * @return {Boolean}
   */
  isDefaultHash : function (hash) {
    return hash === this.fromPageId(config.site.defaultPage);
  },/*}}}*/

};

export default hashTools;


