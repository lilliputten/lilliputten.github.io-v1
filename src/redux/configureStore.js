/**
 * @module configureStore
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.18, 02:42
 * @version 2018.03.19, 23:52
 */

import config from 'config'
import { applyMiddleware, combineReducers, compose, createStore } from 'redux'
import thunk from 'redux-thunk'
import pageReducer from './reducers/pageReducer'

const rootReducer = combineReducers({
  page : pageReducer
});

const noop = () => {};

const enhancer = compose(
  applyMiddleware(thunk),
  // Adding Redux DevTools
  (config.DEBUG && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()) || noop
);

export default function configureStore(initialState = {}) {
  return createStore(rootReducer, initialState, enhancer);
}
