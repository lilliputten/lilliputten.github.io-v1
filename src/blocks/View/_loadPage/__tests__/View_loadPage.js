import React from 'react'

import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'

import reactTools from 'libs/reactTools'

import Enzyme, {mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import config from 'config'
import hashTools from 'libs/hashTools'

import View from 'b:View m:loadPage'

import configureStore from 'redux/configureStore'

const mock = new MockAdapter(axios);

const initialState = window.REDUX_INITIAL_STATE || {};
const store = configureStore(initialState);

Enzyme.configure({ adapter: new Adapter() })

describe('View_loadPage', () => {

  // Create component...
  const wrapper = mount(
    <View loadPage store={store} />
  );
  // Find it...
  const found = wrapper.find('.View');
  // And get instance...
  const instance = reactTools.getDomComponent(found);

  /** beforeAll ** {{{ */
  beforeAll(() => {
  })/*}}}*/
  /** afterAll ** {{{ */
  afterAll(() => {
    wrapper.unmount();
  })/*}}}*/

  /*{{{*/describe('base', () => {

    it('must be found', () => {
      expect(found).toHaveLength(1);
    })

    it('wrapper snapshot', () => {
      expect(wrapper).toMatchSnapshot('wrapper');
    })

    // NOTE: Too expensive snapshot
    // it('match instance snapshot', () => {
    //   expect(instance).toMatchSnapshot('instance');
    // })

    it('state snapshot', () => {
      expect(instance.state).toMatchSnapshot('instance.state');
    })

    it('props snapshot', () => {
      expect(instance.props).toMatchSnapshot('instance.props');
    })

    it('instance of React.Component', () => {
      expect(instance).toBeInstanceOf(React.Component);
    })

    it('default state', () => {
      expect(instance.state).toMatchSnapshot('default state');
      expect(instance.state.mode).toBe(config.site.defaultMode);
    })

  })/*}}}*/

  /*{{{*/describe('getPageState', () => {

    it('is method', () => {
      expect(typeof instance.getPageState).toBe('function');
    })

    it('snapshot', () => {
      expect(instance.getPageState()).toMatchSnapshot('getPageState');
    })

  })/*}}}*/

  /*{{{*/describe('loadPage', () => {

    // Mocking `pageLoader` axios request
    const pageId = 'test/testPage';
    const pageHash = hashTools.fromPageId(pageId);
    const pageUrl = hashTools.toUrl(pageHash, config.site.defaultExt);
    const pageBody = 'page content';
    const pageHtml = '<p>' + pageBody + '</p>\n';

    /** beforeAll ** {{{ */
    beforeAll(() => {

      // Set mocks...
      mock.onAny(pageUrl).reply(200, pageBody);

    })/*}}}*/
    /** afterAll ** {{{ */
    afterAll(() => {

      mock.reset();

    })/*}}}*/


    it('is method', () => {
      expect(typeof instance.loadPage).toBe('function');
    })

    it('returns page data', (done) => {
      instance.loadPage(pageId)
        .then(result => {
          expect(typeof result).toBe('object');
          expect(result).toMatchSnapshot('loadPage result');
          expect(result.body).toBe(pageBody);
          expect(result.html).toBe(pageHtml);
          expect(typeof result.params).toBe('object');
          expect(result.params.pageId).toBe(pageId);

          expect(instance.state).toMatchSnapshot('state after load');
          expect(instance.state.mode).toBe(config.site.contentMode);
          expect(instance.state.page).toBe(pageId);
          // expect(wrapper).toMatchSnapshot('wrapper after load');
        })
        .then(done)
        .catch(done)
    })

  })/*}}}*/

})

