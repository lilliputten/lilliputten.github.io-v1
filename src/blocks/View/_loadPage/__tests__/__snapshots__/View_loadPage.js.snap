// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View_loadPage base default state: default state 1`] = `
Object {
  "mode": "default",
  "page": "default",
}
`;

exports[`View_loadPage base props snapshot: instance.props 1`] = `
Object {
  "dispatch": [Function],
  "loadPage": true,
  "page": "default",
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "storeSubscription": Subscription {
    "listeners": Object {
      "clear": [Function],
      "get": [Function],
      "notify": [Function],
      "subscribe": [Function],
    },
    "onStateChange": [Function],
    "parentSub": undefined,
    "store": Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    },
    "unsubscribe": [Function],
  },
}
`;

exports[`View_loadPage base state snapshot: instance.state 1`] = `
Object {
  "mode": "default",
  "page": "default",
}
`;

exports[`View_loadPage base wrapper snapshot: wrapper 1`] = `
<Connect(View)
  loadPage={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <View
    dispatch={[Function]}
    loadPage={true}
    page="default"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="View View_mode_default"
      style={Object {}}
    >
      <View-ContentWrapper
        mode="default"
      >
        <div
          className="View-ContentWrapper View-ContentWrapper_mode_default"
          style={Object {}}
        >
          Root content 
          <a
            href="#!test/test"
          >
            test
          </a>
        </div>
      </View-ContentWrapper>
    </div>
  </View>
</Connect(View)>
`;

exports[`View_loadPage getPageState snapshot: getPageState 1`] = `
Object {
  "mode": "default",
  "page": "default",
}
`;

exports[`View_loadPage loadPage returns page data: loadPage result 1`] = `
Object {
  "body": "page content",
  "html": "<p>page content</p>
",
  "params": Object {
    "pageId": "test/testPage",
    "pageUrl": "/site/test/testPage.md",
    "paramsUrl": "/site/test/testPage.json",
  },
}
`;

exports[`View_loadPage loadPage returns page data: state after load 1`] = `
Object {
  "html": "<p>page content</p>
",
  "mode": "content",
  "page": "test/testPage",
  "params": Object {
    "pageId": "test/testPage",
    "pageUrl": "/site/test/testPage.md",
    "paramsUrl": "/site/test/testPage.json",
  },
}
`;
